testing/sdk_auth.py

```diff
@@ -244,20 +211,49 @@ def __init__(self):
             kdc_app_def = json.load(f)
 
         kdc_app_def["id"] = KERBEROS_APP_ID
-        _launch_marathon_app(kdc_app_def)
-        _check_kdc_marathon_task_is_running()
+        launch_marathon_app(kdc_app_def)
         self.kdc_port = 88
         self.kdc_fqdn = "{app_id}.marathon.{host_suffix}".format(
             app_id=KERBEROS_APP_ID, host_suffix=sdk_hosts.VIP_HOST_SUFFIX)
-        self.kdc_host_id = _get_host_id()
+        self.kdc_task = _get_kdc_task()
+        self.framework_id = self.kdc_task["framework_id"]
+        self.task_id = self.kdc_task["id"]
+        self.kdc_host_id = self.kdc_task["slave_id"]
         self.kdc_host_name = _get_host_name(self.kdc_host_id)
         self.master_public_ip = _get_master_public_ip()
-        self.framework_id = _get_framework_id()
-        self.task_id = _get_task_id()
         self.principals = []
         self.keytab_file_name = KERBEROS_KEYTAB_FILE_NAME
         self.base64_encoded_keytab_file_name = BASE64_ENCODED_KEYTAB_FILE_NAME.format(keytab_name=self.keytab_file_name)
 
+        # For secret creation/deletion
+        cmd = "package install --yes --cli dcos-enterprise-cli"
+        try:
+            sdk_cmd.run_cli(cmd)
+        except dcos.errors.DCOSException as e:
+            raise RuntimeError("Failed to install the dcos-enterprise-cli: {}".format(repr(e)))
+
+    def __run_kadmin(self, options: list, cmd: str, args: list):
+        """
+        Invokes Kerberos' kadmin binary inside the container to run some command.
+        :param options (list): A list of options given to kadmin.
+        :param cmd (str): The name of the sub command to run.
+        :param args (list): A list of arguments passed to the sub command. This should also include any flags
+                            needed to be set for the sub command.
+        :raises a generic Exception if the invocation fails.
+        """
+        kadmin_cmd = "task exec {task_id} /usr/sbin/kadmin {options} {cmd} {args}".format(
+            task_id=self.task_id,
+            options=' '.join(options),
+            cmd=cmd,
+            args=' '.join(args)
+        )
+        log.info("Running kadmin: {}".format(kadmin_cmd))
+        try:
+            sdk_cmd.run_cli(kadmin_cmd).strip()
```

Comment:
Was there a reason that we chose `run_cli` here instead of `sdk_tasks.task_exec`?

Reply:
Nope, updated.